src/main/java/com/microsoft/spring/data/gremlin/query/GremlinTemplate.java

```diff
@@ -349,44 +328,49 @@ public long edgeCount() {
         return results.size();
     }
 
-    private <T> T recoverDomain(@NonNull GremlinSource source, @NonNull Result result,
-                                @NonNull Class<T> domainClass, boolean isEntityEdge) {
+    private <T> T recoverDomain(@NonNull GremlinSource source, @NonNull Result result) {
```

Comment:
The generic parameter T plays no purpose here either, since the arguments of the method give no indication on `T`. With the current implementation of the `recoverDomain` method, the caller has no way of knowing what the type T is...
This would make sense if the `GremlinSource` interface had a generic type, i.e. `GremlinSource<T>`.